### PR TITLE
Add keyboard ajisai74

### PR DESCRIPTION
**Pull request by Salicylic-acid3**
Add ajisai74 keyboard.

A 74 keys ANSI/JIS Layout keyboard.
Salicylic-acid3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Salicylic-acid3** [comment]
Is there anything wrong with this pull request?

File: keyboards/ajisai74/keymaps/salicylic/keymap.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+extern uint8_t is_master;
+
```
Comment (zvecr) [inline keyboards/ajisai74/keymaps/salicylic/keymap.c:5]:
```suggestion

```

File: keyboards/ajisai74/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#include QMK_KEYBOARD_H
+
+extern uint8_t is_master;
+
```
Comment (zvecr) [inline keyboards/ajisai74/keymaps/via/keymap.c:4]:
```suggestion

```

File: keyboards/ajisai74/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  //|--------+--------+--------+--------+--------+--------|   |--------+--------+--------+--------+--------+--------+--------+--------+--------|
+      _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______,          _______, _______, _______ 
+  //|-----------------------------------------------------|   |--------------------------------------------------------------------------------'
+  )
+};
+
+//A description for expressing the layer position in LED mode.
+layer_state_t layer_state_set_user(layer_state_t state) {
+return state;
+}
```
Comment (zvecr) [inline keyboards/ajisai74/keymaps/via/keymap.c:77]:
```suggestion
```

File: keyboards/ajisai74/rules.mk [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight. 
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (zvecr) [inline keyboards/ajisai74/rules.mk:31]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/ajisai74/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
+#include QMK_KEYBOARD_H
+
+extern uint8_t is_master;
```
Comment (zvecr) [inline keyboards/ajisai74/keymaps/default/keymap.c:3]:
```suggestion
```

File: keyboards/ajisai74/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  //|--------+--------+--------+--------+--------+--------|   |--------+--------+--------+--------+--------+--------+--------+--------+--------|
+      _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______,          _______, _______, _______ 
+  //|-----------------------------------------------------|   |--------------------------------------------------------------------------------'
+  )
+};
+
+//A description for expressing the layer position in LED mode.
+layer_state_t layer_state_set_user(layer_state_t state) {
+return state;
+}
```
Comment (zvecr) [inline keyboards/ajisai74/keymaps/default/keymap.c:47]:
```suggestion
``

File: keyboards/ajisai74/keymaps/jis/keymap.c [hunk 7]
```diff
@@ -0,0 +1,51 @@
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+extern uint8_t is_master;
+
```
Comment (zvecr) [inline keyboards/ajisai74/keymaps/jis/keymap.c:5]:
```suggestion

```

File: keyboards/ajisai74/rev1/rev1.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "ajisai74.h"
```
Comment (zvecr) [inline keyboards/ajisai74/rev1/rev1.h:3]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/ajisai74/rules.mk [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+SPLIT_KEYBOARD = yes
+
+DEFAULT_FOLDER = ajisai74
```
Comment (noroadsleft) [inline keyboards/ajisai74/rules.mk:26]:
```suggestion
```

**Comment by Salicylic-acid3** [comment]
The reason I need tap-dancing in jis and salicylic keymaps is that we want to prevent KC_ENT from being sent twice when the adjacent enter key is pressed at the same time.

File: keyboards/ajisai74/ajisai74.h [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+ * ,------------------------------------   ------------------------------------------------------------.
+ * | L00 | L01 | L02 | L03 | L04 | L05 |   | R00 | R01 | R02 | R03 | R04 | R05 | R06 | R07 | R08 | R18 |
+ * |---------------------------------------------------------------------------------------------------+
+ * |   L10  | L11 | L12 | L13 | L14 | L15 |   | R10 | R11 | R12 | R13 | R14 | R15 | R16 |   R17  | R28 |
+ * |----------------------------------------  ---------------------------------------------------------+
+ * |   L20   | L21 | L22 | L23 | L24 | L25 |   | R20 | R21 | R22 | R23 | R24 | R25 |  R26 |  R27 | R38 |
+ * |---------------------------------------------------------------------------------------------------+
+ * |  L30 | L45 | L31 | L32 | L33 | L34 | L35 |   | R30 | R31 | R32 | R33 | R34 |   R35    | R36 | R37 |
+ * |-------------------------------------------   -----------------------------------------------------+
+ * |  L40  |  L41  |  L42  |  L43  |  L44  |   |  R40 | R41 |  R42 | R43 | R44 | R45 | R46 | R47 | R48 |
```
Comment (noroadsleft) [inline keyboards/ajisai74/ajisai74.h:32]: `R45` doesn't exist.

```suggestion
 * |  L40  |  L41  |  L42  |  L43  |  L44  |   |  R40 | R41 |  R42 | R43 | R44 |     | R46 | R47 | R48 |
```

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**